recipes/opengv/all/conanfile.py

```diff
@@ -54,26 +48,14 @@ def validate(self):
         #FIXME disable this one CCI has more RAM available
         if self.settings.compiler == "gcc" and self.options.shared:
             raise ConanInvalidConfiguration("Shared builds not supported with gcc since CCI errors out due to excessive memory usage.")
+        if self.settings.compiler == "clang" and self.settings.compiler.version == 12:
+            raise ConanInvalidConfiguration("Clang 12 builds fail on Conan CI.")
+
+    def layout(self):
+        cmake_layout(self, src_folder="src")
 
     def source(self):
-        tools.get(**self.conan_data["sources"][self.version],
-                  destination=self._source_subfolder, strip_root=True)
-
-    def _configure_cmake(self):
-        if self._cmake:
-            return self._cmake
-        self._cmake = CMake(self)
-        self._cmake.definitions["BUILD_TESTS"] = False
-        self._cmake.definitions["BUILD_PYTHON"] = self.options.with_python_bindings
-        self._cmake.definitions["BUILD_POSITION_INDEPENDENT_CODE"] = self.settings.os != "Windows" and self.options.get_safe("fPIC", True)
-        if tools.cross_building(self):
-            cmake_system_processor = {
-                "armv8": "aarch64",
-                "armv8.3": "aarch64",
-            }.get(str(self.settings.arch), str(self.settings.arch))
-            self._cmake.definitions["CONAN_OPENGV_SYSTEM_PROCESSOR"] = cmake_system_processor
-        self._cmake.configure()
-        return self._cmake
+        get(self, **self.conan_data["sources"][self.version], destination=self.source_folder, strip_root=True)
 
     def _patch_sources(self):
```

Comment:
Patch files are prefered https://github.com/conan-io/conan-center-index/blob/master/docs/policy_patching.md#patches-format-and-conventions 🙏 if you can sneak that in it would be great

Reply:
Any reason why we prefer patches? I understand how they make the recipe cleaner, but `replace_in_file` is way more flexible.
If the library maintainer adds a single line to the patched files, we need to re do the patches. For a rapid-changing library then we end up with many versioned patches.
That's not the case for this library so I can add the patches though.

Reply:
replace_in_file has no control over the version, we need to open source files and verify. Using patch files, we have some idea about what is changing. Also, we can classify patches on conandata.yml